src/data-types/char.ts

```diff
@@ -27,7 +32,9 @@ module.exports = {
     }
   },
 
-  resolveLength: function(parameter) {
+  //ParameterData<any> is temporary solution. TODO: need to understand what type ParameterData<...> can be. 
+  resolveLength: function(parameter: ParameterData<any>): number {
```

Comment:
I think you can leave off the return type:

```suggestion
  resolveLength: function(parameter: ParameterData<any>) {
```